### DISCUSSION TITLE
Cfclient and sending attitude commands

### DISCUSSION THREAD

**Discussion by mgiglia92** [bitcraze/discussions#302] [Q&A]
Hello,

I'm using crazyflie 2.x as a demo/lab course. The crazyflie is mounted in a 3-axis gimbal system therefore locked from translational motion but free for rotational motion. The course is a "drone controls" course and I'd like to be able to send step input commands as attitude commands either from a separate python script/gui while also being able to watch the log on the cfclient.

The plan would be to adjust PID gains and compare theoretical responses to actual responses. What would be the best approach here? Should I edit the cfclient software and add my own tab that allows for something like this? Or is it possible/easier for me to easily emulate a joystick controller and map commands?

Interested in your thoughts!

Thank You!

**Comment by mgiglia92** [comment]
Okay I got the commands through cf.commander.send_setpoint(). Now i need a way to prevent the cf from shutting down if the rates are too high or if the attitude is upside down. It's in a gimbal so I need it to be allowed to do these non-normal motions. Do I need to change the firmware or are there some parameters I can set to prevent this?

**Reply by knmcguire** [reply]
You should probably look if you can shut off the tumble detection: https://github.com/bitcraze/crazyflie-firmware/blob/b370e20e50106dc10837c97e28404ff5d3efc5ea/src/modules/src/supervisor.c#L95

**Reply by mgiglia92** [reply]
Going to check this out today. Wish me luck.

**Reply by mgiglia92** [reply]
@knmcguire Found what to edit, line 120 in supervisor.c. Was able to flash the firmware using cload.

I'm curious, how do I go about making a .zip file that way I can let my students be able to easily flash firmware using cfclient. Also it will let me have multiple binaries for different use cases that are named to easily distinguish. I'd prefer to have a folder of a bunch of different .zip files of firmware that the students can easily choose from and not have to have a bunch of different git branches for each firmware build and have to use cload each time. Obviously if they're going to be making changes to the on board controller firmware then we will use the process described [https://www.bitcraze.io/documentation/tutorials/getting-started-with-development/#next-steps](url). But for simple things I'd like to have this .zip capability!

Thank you!

**Reply by knmcguire** [reply]
That is possible. You can download the zip file from repo [crazyflie-release](https://github.com/bitcraze/crazyflie-release) and replace the binfiles inside with the one you generated .

We had an old forum post about this too: https://forum.bitcraze.io/viewtopic.php?p=17592&hilit=zip#p17592.

If you have more questions about this, I would advise to start a new discussion as this is technically a different topic than the original post 😉

**Reply by mgiglia92** [reply]
Noted! Thank you!
